### PR TITLE
Add getSessionUser and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Ragudos**
* Add [feature] get user based from session cookie. Fixes: Issue #57

* run prettier

* remove the stored signed authentication cookie if it cannot be deserialized.

* throw err from getSessionUser instead of returning null if the reason is not becuase of a malformed cookie

* fix [feature]: add middleware to delete malformed cookies, separate getSessionUser into a different file since MongoDB is undefined in middleware. Fixes Issue #57

* add CONTRIBUTING.md Fixes: Issue #59

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { NextRequest, NextResponse } from "next/server";
+import {
+  ERROR_INVALID_SESSION_SCHEMA,
+  SESSION_COOKIE_NAME,
+  verifySession,
+} from "./lib/tools/session";
+
+export async function middleware(request: NextRequest) {
+  const sessionCookie = request.cookies.get(SESSION_COOKIE_NAME)?.value;
```
Comment (coltonehrman) [inline src/middleware.ts:9]: Can we not use the `cookies()` function here?

File: src/lib/tools/session/index.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    if (!("userId" in payload)) {
+      throw new Error(ERROR_INVALID_SESSION_SCHEMA);
+    }
+
+    return payload as JWTSessionPayload;
+  } catch (error) {
+    if (error instanceof Error) {
+      if (error.name === "JWSSignatureVerificationFailed") {
+        throw new Error(ERROR_INVALID_SESSION_SCHEMA);
```
Comment (coltonehrman) [inline src/lib/tools/session/index.ts:60]: So we are catching all errors except for this one?

Comment (coltonehrman) [reply src/lib/tools/session/index.ts:60]: Should we just catch everything and always return null on error?

Comment (Ragudos) [reply src/lib/tools/session/index.ts:60]: Well, we are logging the error already, so I thought it wasn't necessary to rethrow it.

File: src/middleware.ts (hunk 1 above)
Comment (Ragudos) [reply src/middleware.ts:9]: Yeah, we can. I'll make a new commit. I thought I couldn't earlier for some reason.

File: src/lib/tools/session/db.ts
Comment (coltonehrman) [inline src/lib/tools/session/db.ts:1]: Maybe call this one `session/api.ts`? Not sure if `db` makes sense. We will eventually have a `model.ts` to define the DB model.

File: src/lib/tools/session/db.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+export type SessionUser = Pick<User, "name" | "roles" | "_id">;
+
+/**
+ * Uses {@link getDeserializedSessionCookie} to get the session token
+ * and then fetches the user from the database.
+ *
+ * Example return response of {@link SessionUser}
+ *
+ * The ```_id``` is a MongoDB ObjectId string.
```
Comment (coltonehrman) [inline src/lib/tools/session/db.ts:17]:
```suggestion
 * The `_id` is a MongoDB ObjectId string.
```

File: src/middleware.ts (hunk 1 above)
Comment (Ragudos) [reply src/middleware.ts:9]: Oh, yep. It's not applicable for the response. We need to modify the response's cookies and for some reason, `clearSessionCookie()` doesn't work for the middleware.

File: src/lib/tools/session/index.ts (hunk 2 above)
Comment (coltonehrman) [reply src/lib/tools/session/index.ts:60]: Right, I wonder if we should re-throw this one either?

File: src/lib/tools/session/db.ts
Comment (Ragudos) [reply src/lib/tools/session/db.ts:1]: Alright

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+## Guidelines
+
+- For commit messages, make sure to follow the discussion [here](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53) as much as possible.
+- Comment or let yourself be assigned to an issue before working on it. **Do not** work on an issue that someone has already commented or been assigned on, unless it is known that they are not working on it anymore.
+- Create a separate branch for the feature/issue you are working on. For example, if you are working on the landing page, you run this command on your terminal or command prompt before starting your work:
+
+```bash
+git checkout -b ui/landing-page
```
Comment (coltonehrman) [inline CONTRIBUTING.md:8]: Maybe make this a bit more generic `git checkout -b feat/<REPLACE_ME>`?
I wonder if someone would get confused here and always use this exact branch name.

File: src/middleware.ts (hunk 1 above)
Comment (Ragudos) [reply src/middleware.ts:9]: The get() function works, so I'm able to clear 3 lines of code.

**Comment by coltonehrman** [comment]
@Ragudos Can we add these rest of the contributing related stuff from README? https://github.com/Software-Developer-Academy/mentor-match?tab=readme-ov-file#clone-the-repository

**Comment by Ragudos** [comment]
Sure

File: CONTRIBUTING.md (hunk 4 above)
Comment (Ragudos) [reply CONTRIBUTING.md:8]: Fair enough

File: src/middleware.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    } catch (err) {
+      if (
+        !(err instanceof Error) ||
+        err.message !== ERROR_INVALID_SESSION_SCHEMA
+      ) {
+        console.error(err);
+        return NextResponse.next();
+      }
+
+      if (request.nextUrl.pathname !== "/signin" && request.method === "GET") {
```
Comment (coltonehrman) [inline src/middleware.ts:23]: Not sure if we need this right now do we? Also, wouldn't it redirect the landing and signup flows?

Comment (Ragudos) [reply src/middleware.ts:23]: What if we are on `/signin`? That'd cause an infinite loop of redirects. I guess I can add the `/` and `/signup` as well to the check.

Comment (coltonehrman) [reply src/middleware.ts:23]: Sorry, I meant the whole redirect logic isn't needed right now, not just the conditional.

Comment (Ragudos) [reply src/middleware.ts]: I see. So, I should just delete the malformed cookie and not redirect.

File: src/lib/tools/session/index.ts (hunk 2 above)
Comment (Ragudos) [reply src/lib/tools/session/index.ts]: Well, that error is thrown when a cookie is malformed.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
+<<<<<<< HEAD
```
Comment (coltonehrman) [inline CONTRIBUTING.md:1]: Looks like you committed some merged conflicts

**Review by coltonehrman — APPROVED** [review]

**Pull request merged**